src/metrics-gatherer.ts

```diff
@@ -28,7 +28,27 @@ const constructors: ConstructorMap = {
 	histogram: new MetricConstructor(prometheus.Histogram),
 };
 
+// decorator function to avoid errors in method calls to the library
```

Comment:
Decorators are an experimental feature and not guaranteed to be supported going forwards (https://www.typescriptlang.org/docs/handbook/decorators.html), so I'd rather not use them in production code to avoid issues

Reply:
Fair point, I'll just add a try catch around each method, then? A bit boiler-plate, but I suppose if it works and the language doesn't have a better feature.